lib/resourceapply/cv.go

```diff
@@ -53,29 +56,33 @@ func ApplyClusterVersionFromCache(lister cvlistersv1.ClusterVersionLister, clien
 	return actual, true, err
 }
 
-func ApplyClusterVersionStatusFromCache(lister cvlistersv1.ClusterVersionLister, client cvclientv1.ClusterVersionsGetter, required *cvv1.ClusterVersion) (*cvv1.ClusterVersion, bool, error) {
-	obj, err := lister.Get(required.Name)
-	if errors.IsNotFound(err) {
-		actual, err := client.ClusterVersions().Create(required)
-		if err != nil && !errors.IsAlreadyExists(err) {
-			return actual, true, err
+// ApplyClusterVersionStatus attempts to overwrite the status subresource of required. If
+// original is provided it is compared to required and no update will be made if the
+// object does not change. The method will retry a conflict by retrieving the latest live
+// version and updating the metadata of required. required is modified if the object on
+// the server is newer.
+func ApplyClusterVersionStatus(client cvclientv1.ClusterVersionsGetter, required, original *cvv1.ClusterVersion) (*cvv1.ClusterVersion, error) {
```

Comment:
Do you want to move this function out to `pkg/cvo`? It might be confusing be see this as other functions use `exsiting, required.`

Reply:
yeah